python/kserve/kserve/protocol/infer_type.py

```diff
@@ -31,7 +31,7 @@ class InferInput:
     _datatype: str
     _parameters: Dict
 
-    def __init__(self, name, shape, datatype, data=None, parameters={}):
+    def __init__(self, name, shape, datatype, data=None, parameters=None):
```

Comment:
We should not use mutable objects in arguments as it can introduce [unintended side effects](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).